alphadia/transferlearning/train.py

```diff
@@ -340,56 +340,58 @@ def _test_ms2(
 
         """
         continue_training = True
-        if epoch % self.settings["test_interval"] == 0:
-            self.ms2_model.model.eval()
-            if "instrument" not in precursor_df.columns:
-                precursor_df["instrument"] = default_instrument
-            if "nce" not in precursor_df.columns:
-                precursor_df["nce"] = default_nce
-
-            precursor_copy = precursor_df.copy()
-            pred_intensities = self.ms2_model.predict(precursor_copy)
-
-            test_input = {
-                "psm_df": precursor_df,
-                "predicted": pred_intensities,
-                "target": target_fragment_intensity_df,
-            }
-            val_metrics = metric_accumulator.calculate_test_metric(
-                test_input, epoch, data_split=data_split, property_name="ms2"
+        if epoch % self.settings["test_interval"] != 0 and epoch != -1:
+            return continue_training
+
+        self.ms2_model.model.eval()
+        if "instrument" not in precursor_df.columns:
+            precursor_df["instrument"] = default_instrument
+        if "nce" not in precursor_df.columns:
+            precursor_df["nce"] = default_nce
+
+        precursor_copy = precursor_df.copy()
+        pred_intensities = self.ms2_model.predict(precursor_copy)
```

Comment:
we should think about making `predict` work on a copy of the `precursor_df` @GeorgWa

Reply:
Yes, good point. This would also involve making the precursor order more predictable. We have a bit of an ugly fix for this at the moment 😄